lib/modules/manager/argocd/extract.ts

```diff
@@ -15,17 +15,66 @@ import { fileTestRegex } from './util';
 
 function createDependency(
   definition: ApplicationDefinition
-): PackageDependency | null {
-  let source: ApplicationSource;
+): Array<PackageDependency | null> {
   switch (definition.kind) {
     case 'Application':
-      source = definition?.spec?.source;
+      if (definition?.spec?.source) {
+        return processMultipleSources([definition?.spec?.source]);
+      }
+      if (definition?.spec?.sources) {
+        return processMultipleSources(definition?.spec?.sources);
+      }
       break;
     case 'ApplicationSet':
-      source = definition?.spec?.template?.spec?.source;
+      if (definition?.spec?.template?.spec?.source) {
+        return processMultipleSources([
+          definition?.spec?.template?.spec?.source,
+        ]);
+      }
+      if (definition?.spec?.template?.spec?.sources) {
+        return processMultipleSources(
+          definition?.spec?.template?.spec?.sources
+        );
+      }
       break;
   }
 
+  return [];
+}
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  _config?: ExtractConfig
+): PackageFileContent | null {
+  // check for argo reference. API version for the kind attribute is used
+  if (fileTestRegex.test(content) === false) {
+    return null;
+  }
+
+  let definitions: ApplicationDefinition[];
+  try {
+    definitions = loadAll(content) as ApplicationDefinition[];
+  } catch (err) {
+    logger.debug({ err, fileName }, 'Failed to parse ArgoCD definition.');
+    return null;
+  }
+
+  const deps: Array<Array<PackageDependency>> = definitions
```

Comment:
```suggestion
  const deps = definitions
```
type should be inferred by compiler